lib/dest/write-contents/write-buffer.js

```diff
@@ -1,14 +1,30 @@
 'use strict';
 
+var iconv = require('iconv-lite');
+
 var fo = require('../../file-operations');
+var getCodec = require('../../codecs');
 
 function writeBuffer(file, optResolver, onWritten) {
+  var encoding = optResolver.resolve('encoding', file);
+  var codec = getCodec(encoding);
+  if (encoding && !codec) {
+    return onWritten(new Error('Unsupported encoding: ' + encoding));
```

Comment:
The only alternative to emitting the error would be to quietly ignore invalid `encoding` values and then proceed as if it were `'utf8'`. But that might cause some nasty bugs, and pretty hard to track down just from the end results (wrong bytes in the buffer).